### PR TITLE
Feature auto focus run if no cont focus

### PR DESCRIPTION
**Pull request by a-szotyori**
If the device's camera doesn't support continuous focus mode we want to make sure, that there was at least one auto-focus run. For this we do an auto-focus run before taking a picture when the user taps the trigger button.

### How to test
_I will add a test for verifying this behaviour with a mock Camera instance. But until then:_

Run the `ginivision/src/test/(...)/internal/camera/api/CameraParametersHelperTest`.
We don't have devices without cont. focus so what can be tested is that it works as before on our devices.
We will ask Katharina for her S3 Mini which doesn't have cont. focus to verify the behaviour.

### Merging
Conflicts, will be merged locally.

### Ticket 
No ticket.

### CODE REVIEW

File: ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @exclude
+ */
+final class CameraParametersHelper {
+
+    private CameraParametersHelper() {
+    }
+
+    static boolean isFocusModeSupported(String focusMode, Camera camera) {
```
Comment (EarlOfEgo) [inline ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java]: even though it's package private, it should use support annotations

File: ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        return camera.getParameters().getSupportedFocusModes().contains(focusMode);
+    }
+
+    static boolean isUsingFocusMode(String focusMode, Camera camera) {
```
Comment (EarlOfEgo) [inline ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java]: same as above, e.g. will crash if camera is null

File: ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        return camera.getParameters().getFocusMode().equals(focusMode);
+    }
+
+    static boolean isFlashModeSupported(String flashMode, Camera camera) {
```
Comment (EarlOfEgo) [inline ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java]: ⬆️ same

File: ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java (hunk 1 above)
Comment (a-szotyori) [inline ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraParametersHelper.java]: Thanks, really good catch! I added the annotations.

File: ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraController.java [hunk 4]
```diff
@@ -41,7 +47,8 @@
 
     private boolean mPreviewRunning = false;
     private AtomicReference<CompletableFuture<Boolean>> mFocusingFuture = new AtomicReference<>();
-    private AtomicReference<CompletableFuture<Photo>> mTakingPictureFuture = new AtomicReference<>();
```
Comment (nsobadjiev) [inline ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraController.java]: I really wish the commit wasn't full of those line breaks. It's distracting and pushes attention away from the important changes. I'd prefer introducing line length limits to be a separate commit.

**Comment by a-szotyori** [comment]
Added tests for verifying the new behaviour.
